homework2/task01/project1/task1.py

```diff
@@ -0,0 +1,94 @@
+"""
+Given a file containing text. Complete using only default collections.
+
+1) Find 10 longest words consisting from largest amount of unique symbols
+2) Find rarest symbol for document
+3) Count every punctuation char
+4) Count every non ascii char
+5) Find most common non ascii char for document
+"""
+from typing import List
+
+
+def file_presettings(file_path: str) -> str:
+    """
+    Return unpacked file, encode in readable format.
+
+    First sub() deletes '\n' between lines. Second sub() remove punctuation.
+    """
+    with open(file_path, "r", encoding="unicode_escape") as f:
+        text = f.read()
+        for p in '!"#$%&»«()*+,-./:;<=>?@_`{|}~':
+            if p in text:
+                text = text.replace(p, "")
+        return text
+
+
+def get_unique_word_symbols(i: str) -> (str, str, int):
+    """Return key function for sorting."""
+    return len("".join(set(i))), i[0], len(i)
```

Comment:
Please describe each key element in terms of task requirements.

Reply:
Done. I also need i[0] here for sorting to make assertion in test. If I sort only in terms of len("".join(set(i))), I would get not the same 10 words in every testing. Words randomly replaced without sorting alphabetically. Third feature is no need, i deleted it.